tests/cupyx_tests/scipy_tests/special_tests/test_logsumexp.py

```diff
@@ -0,0 +1,108 @@
+    def test_array_inputs(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind in 'u':
+            pytest.skip()
+        a = testing.shaped_random((100, 1000), xp, dtype=dtype)
+        return scp.special.logsumexp(a)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_sign_argument(self, xp, scp, dtype):
+        a = xp.array([1, 1, 1], dtype=dtype)
+        b = xp.array([1, -1, -1], dtype=dtype)
+        return scp.special.logsumexp(a, b=b, return_sign=True)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_sign_zero(self, xp, scp, dtype):
+        a = xp.array([1, 1], dtype=dtype)
+        b = xp.array([1, -1], dtype=dtype)
+        return scp.special.logsumexp(a, b=b, return_sign=True)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', rtol=1e-6)
+    def test_sign_multi_dims(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind in 'u':
+            pytest.skip()
+        a = testing.shaped_random((1, 1, 3, 4), xp, dtype=dtype)
+        b = testing.shaped_random((1, 1, 1, 4), xp, dtype=dtype)
+        return scp.special.logsumexp(a, b=b, return_sign=True)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', rtol=1e-6)
+    def test_sign_shape_axis(self, xp, scp, dtype):
+        a = testing.shaped_random((1, 2, 3, 4), xp, dtype=dtype)
+        b = testing.shaped_random((1, 2, 3, 4), xp, dtype=dtype)
+        return scp.special.logsumexp(a, axis=2, b=b, return_sign=True)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', rtol=1e-6)
+    def test_sign_shape_axis_2d(self, xp, scp, dtype):
+        a = testing.shaped_random((1, 2, 3, 4), xp, dtype=dtype)
+        b = testing.shaped_random((1, 2, 3, 4), xp, dtype=dtype)
+        return scp.special.logsumexp(a, axis=(1, 3), b=b, return_sign=True)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_b_zero(self, xp, scp, dtype):
+        a = xp.array([1, 100], dtype=dtype)
+        b = xp.array([1, 0], dtype=dtype)
+        return scp.special.logsumexp(a, b=b)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_b_shape(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind in 'u':
+            pytest.skip()
+        a = testing.shaped_arange((4, 1, 2, 1), xp, dtype=dtype)
+        b = testing.shaped_arange((3, 1, 5), xp, dtype=dtype)
+        return scp.special.logsumexp(a, b=b)
+
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_special_values(self, xp, scp):
+        a = xp.array([cupy.inf, -cupy.inf, cupy.nan, -cupy.nan])
+        return scp.special.logsumexp(a)
+
+    @testing.for_all_dtypes(no_bool=True)
+    def test_empty_array_inputs(self, dtype):
+        import scipy.special  # NOQA
```

Comment:
We conventionally do not use `import` in a function. Please import it on the top section of this file as:

https://github.com/cupy/cupy/blob/983d2b84e546428b0343151e0187275b5a35a0e9/tests/cupyx_tests/scipy_tests/linalg_tests/test_special_matrices.py#L6-L9